myning/chapters/mine/actions/combat.py

```diff
@@ -0,0 +1,255 @@
+            next_combat_action,
+        )
+
+
+@dataclass
+class RoundLog:
+    attacker: Character
+    defender: Character
+    is_friendly: bool
+    damage: int
+    dodged: bool
+    critted: bool
+
+
+class RoundAction(Action):
+    def __init__(
+        self,
+        bonus: float,
+        damage_done: int,
+        health_lost: int,
+        round_logs: list[RoundLog],
+        next_action: Action,
+    ):
+        damage_width = max(len(str(damage_done)), len(str(health_lost)))
+        self.summary = "[bold]Round Summary[/]\n\n"
+        if not settings.mini_games_disabled:
+            self.summary += f"{_bonus_str(bonus)}\n\n"
+        self.summary += (
+            f"[bold green1]{damage_done:{damage_width}}[/] damage inflicted.\n"
+            f"[bold red1]{health_lost:{damage_width}}[/] damage sustained.\n"
+        )
+        self.log_table = Table.grid(padding=(0, 1, 0, 0))
+        if round_logs:
+            for log in round_logs:
+                self.log_table.add_row(
+                    str(log.attacker.icon),
+                    f"[{'green1' if log.is_friendly else 'red1'}]{log.attacker.name}[/]",
+                    Icons.CRIT if log.critted else "",
+                    "[bold dark_cyan](0)[/]"
+                    if log.dodged
+                    else f"[{'bold orange1' if log.critted else 'normal'}]{log.damage}[/]",
+                    Icons.DODGE if log.dodged else "",
+                    str(log.defender.icon),
+                    f"[{'red1' if log.is_friendly else 'green1'}]{log.defender.name}[/]",
+                    Icons.DEATH if log.defender.health <= 0 else "",
+                )
+            self.log_table.columns[3].justify = "right"
+        self.next_action = next_action
+        super().__init__(5)
+
+    @property
+    def content(self):
+        table = Table.grid(expand=True)
+        table.add_row(self.summary)
+        table.add_row(f"Returning to battle in {self.duration}...\n")
+        table.add_row(self.log_table)
+        return table
+
+    @property
+    def next(self):
+        return self.next_action
+
+
+class VictoryAction(Action):
+    def __init__(self, enemy_count: int):
+        TabTitle.change_tab_subactivity("")
+        assert trip.mine
+        rewards = generate_reward(trip.mine.max_item_level, enemy_count)
+        trip.add_items(*rewards)
+        FileManager.multi_save(trip, *rewards)
+        self.rewards = rewards
+        super().__init__(len(rewards) + 1)
+
+    @property
+    def content(self):
+        lines = ["[green1]You won the battle![/]\n"] + [
+            item.battle_new_str for item in self.rewards[: len(self.rewards) - self.duration + 1]
+        ]
+        return "\n".join(lines)
+
+    @property
+    def next(self):
+        return self if self.duration > 1 else None
+
+
+def _get_battle_order(allies: list[Character], enemies: list[Character]):
+    combined = allies + enemies
+    random.shuffle(combined)
+    return combined
```

Comment:
I think I probably originally wrote this, but I think this can be 1-2 lines less haha

Reply:
`random.shuffle` shuffles in place and returns None so we can't change this